src/charm.py

```diff
@@ -301,7 +301,44 @@ def _build_pod_spec(self):
                         'content': self._prometheus_config()
                     }]
                 }]
-            }]
+            }],
+            'serviceAccount': {
+                'roles': [{
```

Comment:
It may be a good idea to have the value of `serviceAccount` generated and returned by a function. For example like Prometheus configuration is returned by `self._prometheus_config()`. What do you think ?

Reply:
Just did, what do u think?

Reply:
Looks good to me.